dart-impl/CMakeLists.txt

```diff
@@ -17,9 +17,11 @@ add_subdirectory(base)
 
 # DART implementation of every enabled DART variant:
 if (";${DART_IMPLEMENTATIONS_LIST};" MATCHES ";mpi;")
+  if(NOT MPI_IS_DART_COMPATIBLE)
+    message(FATAL_ERROR "Detected MPI library does not support MPI-3.")
+  endif()
```

Comment:
would also be possible. Currently the build system of DASH and DART is quite tightly coupled, unfortunately.